recipes/pcl/all/conanfile.py

```diff
@@ -0,0 +1,318 @@
+    description = "Point Cloud Library"
+    license = "BSD-3-Clause"
+    homepage = "https://pointclouds.org/"
+    url = "https://github.com/conan-io/conan-center-index"
+    topics = ("pointcloud", "computer-vision", "point-cloud")
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_openmp": [True, False],
+        "with_png": [True, False],
+        "with_qhull": [True, False],
+        "with_cuda": [True, False],
+        "with_tools": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_openmp": False,
+        "with_png": True,
+        "with_qhull": True,
+        "with_cuda": False,
+        "with_tools": False,
+    }
+
+    exports = ["CMakeLists.txt"]
+    generators = ["cmake", "cmake_find_package", "cmake_find_package_multi"]
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def _check_msvc(self):
+        if (tools.msvs_toolset(self) == "v140" or
+                self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version) < "15"):
+            raise ConanInvalidConfiguration("Unsupported Visual Studio Compiler or Toolset")
+
+    def _check_cxx_standard(self):
+        minimal_cpp_standard = "14"
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, minimal_cpp_standard)
+        minimal_version = {
+            "gcc": "5",
+            "clang": "3.4",
+            "apple-clang": "10",
+            "Visual Studio": "14"
+        }
+        compiler = str(self.settings.compiler)
+        if compiler not in minimal_version:
+            self.output.warn(
+                "%s recipe lacks information about the %s compiler standard version support" % (self.name, compiler))
+            self.output.warn(
+                "%s requires a compiler that supports at least C++%s" % (self.name, minimal_cpp_standard))
+            return
+        version = tools.Version(self.settings.compiler.version)
+        if version < minimal_version[compiler]:
+            raise ConanInvalidConfiguration("%s requires a compiler that supports at least C++%s" % (self.name, minimal_cpp_standard))
+
+    def _check_libcxx_compatibility(self):
+        if self.settings.compiler == "clang" and self.settings.compiler.libcxx == "libc++":
+            version = tools.Version(self.settings.compiler.version)
+            minimum_version = 6
+            if version < minimum_version:
+                raise ConanInvalidConfiguration("Clang with libc++ is version %s but must be at least version %s" %
+                        (version, minimum_version))
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        self._check_msvc()
```

Comment:
I tested on macOS. This line needed to prevent fail.
```suggestion
        if self.settings.compiler=="msvc":
            self._check_msvc()
```

Reply:
After that getting error. Not sure what is wrong.
<img width="463" alt="Screenshot 2023-02-28 at 21 40 12" src="https://user-images.githubusercontent.com/110199/221987162-56a66b7f-2229-4b26-ac2f-22c024b9898e.png">

Reply:
@sakrist The error on that picture is different, it shows that the package version that you passed to `conan create` command does not exist in `conandata.yml` file. Please, share your entire build log as pure text.

Reply:
True. This is new error I get after fix in first comment. 
oh, I try build 1.12.0 but available only 1.12.1

Reply:
Successfully compiled 1.12.1 package on MacBook M1 Pro.